src/types/dict.rs

```diff
@@ -357,6 +366,24 @@ impl<'py> PyDictMethods<'py> for Bound<'py, PyDict> {
         BoundDictIterator::new(self.clone())
     }
 
+    #[cfg(all(Py_GIL_DISABLED, not(feature = "nightly")))]
+    fn locked_for_each<F>(&self, f: F) -> PyResult<()>
+    where
+        F: Fn(Bound<'py, PyAny>, Bound<'py, PyAny>) -> PyResult<()>,
+    {
+        #[cfg(feature = "nightly")]
+        {
+            self.iter().try_for_each(|(key, value)| f(key, value))
```

Comment:
Maybe this is supposed to be calling `try_fold` on nightly?

Reply:
Oh wait, I see, it's becaues `try_for_each` is implemented in terms of `try_fold` (sorry for all the noise here!).

Can you add a comment explaining the difference?

Reply:
You got it. I will add a comment to prevent confusion next time.